starknet_py/net/full_node_client.py

```diff
@@ -57,10 +58,16 @@ def __init__(
 
         :param node_url: Url of the node providing rpc interface
         :param net: StarkNet network identifier
-        :param chain: Chain id of the network used by the rpc client
+        :param chain: Chain id of the network used by the rpc client. Chain is deprecated
+                        and will be removed in the next releases
         :param session: Aiohttp session to be used for request. If not provided, client will create a session for
-                        every request. When using a custom session, user is resposible for closing it manually.
+                        every request. When using a custom session, user is responsible for closing it manually.
         """
+        if chain is not None:
+            warnings.warn(
+                "Chain is deprecated and will be deleted in the next releases"
+            )
```

Comment:
Add a `DeprecationWarning` category to the warning.

Reply:
Also the internet suggests adding `stacklevel=2` to make warning more readable but I haven't actually verified that so you can test yourself if it looks better with or without this parameter.

Reply:
I can't see a difference with stacklevel=2